pkgs/build-support/mkshell/default.nix

```diff
@@ -54,9 +54,7 @@ stdenv.mkDerivation (
       lib.catAttrs "shellHook" (lib.reverseList inputsFrom ++ [ attrs ])
     );
 
-    phases = [ "buildPhase" ];
-
-    buildPhase = ''
+    buildCommand = ''
```

Comment:
The existence of a `buildCommand` turns off phase execution in `genericBuild` altogether.

https://github.com/NixOS/nixpkgs/blob/b47e6c97edba72ee20b27a26bbcea363409f9c5d/pkgs/stdenv/generic/setup.sh#L1779-L1786

The underlying goal of https://github.com/NixOS/nixpkgs/issues/28910 is to make the execution of phases more regular by default.

Considering that this is not just a derivation, but a helper function that produces derivations (or shells with stdenv builder functionality), I don't know if that's what we're achieving with this.

Specifically, if someone has a `mkShell` shell where they've restored the phases (in `// rest` below), they will find that `genericBuild` in their shell has stopped working.